libtransact/src/contract/archive/mod.rs

```diff
@@ -189,13 +189,30 @@ impl SmartContractArchive {
                 file_path.display()
             ))
         })?;
-        let metadata = metadata.ok_or_else(|| {
+        let metadata: SmartContractMetadata = metadata.ok_or_else(|| {
             Error::new(&format!(
                 "invalid scar file: manifest.yaml not found in {}",
                 file_path.display()
             ))
         })?;
 
+        let version_req = VersionReq::parse(version)?;
+        let metadata_version = Version::parse(&metadata.version).map_err(|err| {
+            Error::new_with_source(
+                &format!(
+                    "invalid metadata version: {}, unable to parse into semver string",
```

Comment:
Can probably just remove `, unable to parse into semver string` since this information should be provided by the source error.

Reply:
Addressed